pkg/propertyprovider/aks/trackers/trackers.go

```diff
@@ -0,0 +1,570 @@
+			tc.Sub(c)
+			nt.totalCapacity[rn] = tc
+		}
+		delete(nt.capacityByNode, nodeName)
+		klog.V(4).InfoS("Untracked the node's total capacity", "node", nodeName)
+	}
+}
+
+// untrackAllocatableCapacity untracks the allocatable capacity of a node.
+func (nt *NodeTracker) untrackAllocatableCapacity(nodeName string) {
+	ra, ok := nt.allocatableByNode[nodeName]
+	if ok {
+		for _, rn := range supportedResourceNames {
+			a := ra[rn]
+			ta := nt.totalAllocatable[rn]
+			ta.Sub(a)
+			nt.totalAllocatable[rn] = ta
+		}
+
+		delete(nt.allocatableByNode, nodeName)
+		klog.V(4).InfoS("Untracked the node's allocatable capacity", "node", nodeName)
+	}
+}
+
+// Remove stops tracking a node.
+func (nt *NodeTracker) Remove(nodeName string) {
+	nt.mu.Lock()
+	defer nt.mu.Unlock()
+
+	// Untrack the total and allocatable capacity of the node.
+	nt.untrackTotalCapacity(nodeName)
+	nt.untrackAllocatableCapacity(nodeName)
+
+	// Untrack the node's SKU.
+	nt.untrackSKU(nodeName)
+
+	// Re-calculate costs.
+	//
+	// Note that it would be very rare for the informer to receive a node deletion event before
+	// having a chance to track it first.
+	nt.calculateCosts()
+}
+
+// NodeCount returns the node count stat that a node tracker tracks.
+func (nt *NodeTracker) NodeCount() int {
+	nt.mu.RLock()
+	defer nt.mu.RUnlock()
+
+	// capacityByNode and allocatableByNode should always have the same length.
+	return len(nt.allocatableByNode)
+}
+
+// TotalCapacityFor returns the total capacity of a specific resource that the node
+// tracker tracks.
+func (nt *NodeTracker) TotalCapacityFor(rn corev1.ResourceName) resource.Quantity {
+	nt.mu.RLock()
+	defer nt.mu.RUnlock()
+
+	return nt.totalCapacity[rn]
+}
+
+// TotalAllocatableFor returns the total allocatable capacity of a specific resource that the node
+// tracker tracks.
+func (nt *NodeTracker) TotalAllocatableFor(rn corev1.ResourceName) resource.Quantity {
+	nt.mu.RLock()
+	defer nt.mu.RUnlock()
+
+	return nt.totalAllocatable[rn]
+}
+
+// TotalCapacity returns the total capacity of all resources that the node tracker tracks.
+func (nt *NodeTracker) TotalCapacity() corev1.ResourceList {
+	nt.mu.RLock()
+	defer nt.mu.RUnlock()
+
+	return nt.totalCapacity
+}
+
+// TotalAllocatable returns the total allocatable capacity of all resources that
+// the node tracker tracks.
+func (nt *NodeTracker) TotalAllocatable() corev1.ResourceList {
+	nt.mu.RLock()
+	defer nt.mu.RUnlock()
+
+	return nt.totalAllocatable
+}
+
+// Costs returns the per CPU core and per GB memory costs in the cluster.
+func (nt *NodeTracker) Costs() (perCPUCoreCost, perGBMemoryCost float64, err error) {
+	nt.mu.Lock()
+	defer nt.mu.Unlock()
+	if nt.costLastUpdated.Before(nt.pricingProvider.LastUpdated()) {
+		nt.calculateCosts()
+	}
+	return nt.perCPUCoreHourlyCost, nt.perGBMemoryHourlyCost, nt.costErr
+}
+
+// NodeTracker helps track specific stats about nodes in a Kubernetes cluster, e.g., the sum
+// of its requested resources.
+type PodTracker struct {
```

Comment:
can we move it into a separate file, podtracker.go?

Reply:
Done.